pkg/gitreceive/build.go

```diff
@@ -183,14 +183,19 @@ func build(
 		return fmt.Errorf("creating builder pod (%s)", err)
 	}
 
-	if err := waitForPod(kubeClient, newPod.Namespace, newPod.Name, conf.SessionIdleInterval(), conf.BuilderPodTickDuration(), conf.BuilderPodWaitDuration()); err != nil {
+	pw := k8s.NewPodWatcher(kubeClient, "deis")
```

Comment:
Right now we just need it during the slugbuilder/dockerbuilder pod which happens only during a new build which is not very frequent i assume.....Do we want Controller to be running continuously even when we may not need it?